Remotion/Web/ClientScript/Scripts/ExecutionEngine.ts

```diff
@@ -253,29 +253,29 @@ class WxePage_Context
 
     WxePage_Context._instance = instance;
   }
-}
 
-function WxePage_OnLoading(hasSubmitted: boolean, isCached: boolean, isAsynchronous: boolean): void
-{
-  WxePage_Context.GetInstance()!.OnLoading(hasSubmitted, isCached, isAsynchronous);
-}
+  public static OnLoading (hasSubmitted: boolean, isCached: boolean, isAsynchronous: boolean): void
```

Comment:
Question: we pass the function name in C#
`SmartPage.RegisterClientSidePageEventHandler(...,"WxeContext.OnLoading") and then use `fct = eval(functionName);` in JavaScript to get the function for this string. Does this also work for dotted function names?

Reply:
With `eval("WxeContext.OnLoading")` it does, with something like `window["WxeContext.OnLoading"]` it wouldn't

Reply:
ah, okay. Could you please consider if there's more elegant ways to write this event dispatching logic and if those would require the global function syntax? If there's none, we can keep the new logic.

Reply:
I don't think that there is a more elegant way to do this. But it would probably be a good idea to get rid of the `eval` and use a custom resolver that looks up the function name from the window object.

Reply:
As discussed, we should keep the gobal-function requirement here. It keeps future code simpler and we can always change our minds later on.